spring-cloud-generator/scripts/generate-showcase.sh

```diff
@@ -0,0 +1,105 @@
+# $2 - directory containing newly generated showcase-spring-starter
+function verify(){
+  OLD_DIR=$1
+  NEW_DIR=$2
+  SHOWCASE_STARTER_DIFF=$(diff -r ${NEW_DIR}/src/main ${OLD_DIR}/src/main)
+  SHOWCASE_STARTER_POM_DIFF=$(diff -r ${NEW_DIR}/pom.xml ${OLD_DIR}/pom.xml)
+  if [ "$SHOWCASE_STARTER_DIFF" != "" ] || [ "$SHOWCASE_STARTER_POM_DIFF" != "" ]
+  then
+      echo "Differences detected in generated showcase starter module: "
+      echo "Diff from src/main: "
+      echo $SHOWCASE_STARTER_DIFF
+      echo "Diff from pom.xml: "
+      echo $SHOWCASE_STARTER_POM_DIFF
+      exit 1;
+  else
+      echo "No differences found in showcase-spring-starter"
+      rm -r ${NEW_DIR}
+  fi
+}
+
+# Setup, generation, and post-processing steps for showcase-spring-starter
+#
+# $1 - target directory for generated starter
+function generate_showcase_spring_starter(){
+  SHOWCASE_STARTER_DIR=$1
+
+  # Compute the parent project version.
+  cd ${SPRING_ROOT_DIR}
+  PROJECT_VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)
+  cd ${SPRING_GENERATOR_DIR}
+  GAPIC_GENERATOR_JAVA_VERSION=$(./../mvnw help:evaluate -Dexpression=gapic-generator-java-bom.version -q -DforceStdout)
+
+  # Clone sdk-platform-java (with showcase library)
+  git clone https://github.com/googleapis/sdk-platform-java.git
+  git checkout "v${GAPIC_GENERATOR_JAVA_VERSION}"
+
+  # Install showcase client libraries locally
+  cd sdk-platform-java && mvn clean install -B -ntp -DskipTests -Dclirr.skip -Dcheckstyle.skip
+  cd showcase && mvn clean install
+
+  # Alternative: if showcase client library is available on Maven Central,
+  # Instead of downloading sdk-platform-java/showcase (for client library, and generation setup),
+  # Can instead download googleapis (for generation setup) and gapic-showcase (for protos)
+
+  # Modify sdk-platform-java/WORKSPACE
+  modify_workspace_file "../WORKSPACE" ".." "../../scripts/resources/googleapis_modification_string.txt"
+  # Modify sdk-platform-java/showcase/BUILD.bazel
+  buildozer 'new_load @spring_cloud_generator//:java_gapic_spring.bzl java_gapic_spring_library' BUILD.bazel:__pkg__
+  modify_build_file "BUILD.bazel"
+
+  # Invoke bazel target for generating showcase-spring-starter
+  bazelisk build --tool_java_language_version=17 --tool_java_runtime_version=remotejdk_17 //showcase:showcase_java_gapic_spring
+
+  # Post-process generated modules
+  copy_and_unzip "../bazel-bin/showcase/showcase_java_gapic_spring-spring.srcjar" "showcase_java_gapic_spring-spring.srcjar" "${SPRING_GENERATOR_DIR}/showcase" ${SHOWCASE_STARTER_DIR}
+  modify_starter_pom ${SHOWCASE_STARTER_DIR}/pom.xml "com.google.cloud" "gapic-showcase" $PROJECT_VERSION
+
+  # Additional pom.xml modifications for showcase starter
+  # Add explicit gapic-showcase version
+  sed -i '/^ *<artifactId>gapic-showcase<\/artifactId>*/a \ \ \ \ \ \ <version>0.0.1-SNAPSHOT</version>' ${SHOWCASE_STARTER_DIR}/pom.xml
```

Comment:
Please add a line to read the showcase version (using above `mvnw help:evaluate` syntax) rather than assuming `0.0.1-SNAPSHOT`

Reply:
Will do. (Memo: should correspond to version [here](https://github.com/googleapis/sdk-platform-java/blob/1ace49499175f2509062edb2fa5098e6cf8035d6/showcase/gapic-showcase/pom.xml#L8))

Reply:
Updated in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1957/commits/2b6a0701ecabc02a51ba27ddb08561cce0682935.